MastodonSDK/Sources/MastodonUI/View/Content/StatusMetricRowView.swift

```diff
@@ -26,48 +28,92 @@ public final class StatusMetricRowView: UIButton {
+            @unknown default:
+                break
+            }
+        }
+    }
+
+    var margin: CGFloat = 0 {
+        didSet {
+            layoutMargins = UIEdgeInsets(horizontal: margin, vertical: 0)
+        }
+    }
+
     private func setupConstraints() {
+        icon.setContentHuggingPriority(.defaultHigh, for: .horizontal)
+        chevron.setContentHuggingPriority(.defaultHigh, for: .horizontal)
         let constraints = [
             icon.topAnchor.constraint(greaterThanOrEqualTo: topAnchor, constant: 10),
-            icon.leadingAnchor.constraint(equalTo: leadingAnchor),
+            icon.leadingAnchor.constraint(equalTo: layoutMarginsGuide.leadingAnchor),
             icon.centerYAnchor.constraint(equalTo: centerYAnchor),
-            textLabel.leadingAnchor.constraint(equalTo: icon.trailingAnchor, constant: 16),
+            contentStack.leadingAnchor.constraint(equalTo: icon.trailingAnchor, constant: 16),
             icon.widthAnchor.constraint(greaterThanOrEqualToConstant: 24),
             icon.heightAnchor.constraint(greaterThanOrEqualToConstant: 24),
             bottomAnchor.constraint(greaterThanOrEqualTo: icon.bottomAnchor, constant: 10),
 
-            textLabel.topAnchor.constraint(greaterThanOrEqualTo: topAnchor, constant: 11),
-            detailLabel.leadingAnchor.constraint(greaterThanOrEqualTo: textLabel.trailingAnchor, constant: 8),
-            textLabel.centerYAnchor.constraint(equalTo: centerYAnchor),
-            bottomAnchor.constraint(greaterThanOrEqualTo: textLabel.bottomAnchor, constant: 11),
-
-            detailLabel.topAnchor.constraint(greaterThanOrEqualTo: topAnchor, constant: 11),
-            chevron.leadingAnchor.constraint(greaterThanOrEqualTo: detailLabel.trailingAnchor, constant: 12),
-            detailLabel.centerYAnchor.constraint(equalTo: centerYAnchor),
-            bottomAnchor.constraint(greaterThanOrEqualTo: detailLabel.bottomAnchor, constant: 11),
+            contentStack.topAnchor.constraint(greaterThanOrEqualTo: topAnchor, constant: 11),
+            contentStack.centerYAnchor.constraint(equalTo: centerYAnchor),
+            bottomAnchor.constraint(greaterThanOrEqualTo: contentStack.bottomAnchor, constant: 11),
+            chevron.leadingAnchor.constraint(equalTo: contentStack.trailingAnchor, constant: 12),
 
             chevron.centerYAnchor.constraint(equalTo: centerYAnchor),
-            trailingAnchor.constraint(equalTo: chevron.trailingAnchor),
+            layoutMarginsGuide.trailingAnchor.constraint(equalTo: chevron.trailingAnchor),
         ]
 
         NSLayoutConstraint.activate(constraints)
     }
+
+    public override var isHighlighted: Bool {
+        get { super.isHighlighted }
+        set {
+            super.isHighlighted = newValue
+            if newValue {
+                backgroundColor = Asset.Colors.selectionHighlight.color
+            } else {
+                backgroundColor = .clear
+            }
+        }
+    }
```

Comment:
Nice workaround :)

Reply:
Thanks! It’s not 100% because it doesn’t do the nice fade out when you slowly swipe back from the pushed view controller like a real table cell does, but it’s close enough for now!